packages/amazonq/test/unit/codewhisperer/util/codeParsingUtil.test.ts

```diff
@@ -106,42 +112,48 @@ describe('isTestFile', () => {
         }
     })
 
-    async function assertIsTestFile(srcFiles: string[], tstFiles: string[], fileExt: string) {
-        for (const name of tstFiles) {
-            const file = `${name}.${fileExt}`
-            const editor = await openATextEditorWithText('', file, tempFolder, { preview: false })
-            const actual = await isTestFile(editor.document.uri.fsPath, { languageId: editor.document.languageId })
-            assert.strictEqual(actual, true)
-        }
-
-        for (const name of srcFiles) {
-            const file = `${name}.${fileExt}`
-            const editor = await openATextEditorWithText('', file, tempFolder, { preview: false })
-            const actual = await isTestFile(editor.document.uri.fsPath, { languageId: editor.document.languageId })
-            assert.strictEqual(actual, false)
+    async function assertIsTestFile(
+        filePaths: string[],
+        config: { languageId: PlatformLanguageId },
+        expected: boolean
+    ) {
+        for (const filePath of filePaths) {
+            const editor = createMockTextEditor('', filePath, config.languageId)
+            const actual = await isTestFile(editor.document.uri.fsPath, { languageId: config.languageId })
```

Comment:
isTestFile only uses the document, so why is an editor needed?

Reply:
that's a good callout, i should update the signature to use document rather than an editor

Reply:
why create a *mock*? my [suggestion](https://github.com/aws/aws-toolkit-vscode/pull/5590#discussion_r1768792061) was simply to skip opening an *editor*. Using `vscode.workspace.openTextDocument` to create a real document (without opening an editor) should be fast, which avoids the CI slowness. That doesn't require using a mock.

Reply:
isee